### ISSUE TITLE
tearDown/setUp considered an invalid name for classes inherited indirectly from TestCase

### ISSUE THREAD

**Issue by socketpair** [PyCQA/pylint#4060]
```
C0103: Method name "tearDown" doesn't conform to '^(test_)?[a-z_][a-z0-9_]{2,30}$' pattern ('^(test_)?[a-z_][a-z0-9_]{2,30}$' pattern) (invalid-name)
```

I have:

```
method-rgx=^(test_)?[a-z_][a-z0-9_]{2,30}$
```

Seems because pylint does not recognize `IsolatedAsyncioTestCase` in same way as `TestCase`

**Comment by socketpair** [comment]
@hippo91

**Comment by Pierre-Sassoulas** [comment]
I would advise to modify the regex in the configuration:

```
[BASIC]

# Regular expression matching correct method names
# Pylint do not complain for setUp and other if it
# detects that we inherit from unittest.TestCase
# But that's not always possible to detect.
method-rgx=[a-z_][a-z0-9_]{2,30}$|setUp(Class)?|tearDown(Class)?
```

**Comment by hippo91** [comment]
@Pierre-Sassoulas  it makes sense. Thank you. May we close the issue?

**Comment by Pierre-Sassoulas** [comment]
This is a real problem in pylint but I don't think it's going to be handled any time soon. If it is, it will probably be in a very generic issue involving astroid guessing the right inherited class (?). I wouldn't mind closing it. Maybe we could fix this by updating the default regex directly inside pylnt ?

**Comment by hippo91** [comment]
@socketpair can you give a snippet please?
So that i can evaluate the amount of work to be done to handle this.
Thanks.

**Comment by socketpair** [comment]
```python
class BaseTestEtcd(IsolatedAsyncioTestCase):
    @classmethod
    def tearDownClass(cls):
        ...

    def setUp(self):
        ...
```

**Comment by socketpair** [comment]
@hippo91

**Comment by hippo91** [comment]
@socketpair thanks for the snippet. I will work on it ASAP.

**Comment by socketpair** [comment]
@hippo91 Do you have any progress on it ?

**Comment by hippo91** [comment]
@socketpair i am sorry i did not have time to work on this.

**Comment by hippo91** [comment]
@socketpair i investigated a bit the problem. 
This is due to the way `IsolatedAsyncioTestCase` is imported through `unittest` (dynamic lazy import). I will work for a solution ASAP. Meanwhile you can replace `from unittest import IsolatedAsyncioTestCase` with `from unittest.async_case import IsolatedAyncioTestCase`.
